platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/internal/ConcurrentTaskScheduler.java

```diff
@@ -74,18 +76,27 @@ public ConcurrentTaskScheduler(
      */
     @Override
     protected void put(@NonNull final Consumer<Object> handler, @NonNull final Object data) {
+        if (getSquelcher().shouldSquelch()) {
+            return;
+        }
+
         onRamp.onRamp();
-        new ConcurrentTask(pool, offRamp, uncaughtExceptionHandler, handler, data).send();
+        new ConcurrentTask(pool, offRamp, uncaughtExceptionHandler, handler, data, getSquelcher()).send();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     protected boolean offer(@NonNull final Consumer<Object> handler, @NonNull final Object data) {
-        boolean accepted = onRamp.attemptOnRamp();
+        if (getSquelcher().shouldSquelch()) {
+            // say that the data was accepted, even though it will be thrown into the void
+            return true;
+        }
```

Comment:
At first glance this seemed strange, but when I thought about it I think it makes sense. We return true because we want to accept and discard the data, as opposed to preventing the sender from sending it. Might be nice to expand this comment a little to explain.